client/src/app/pages/project-settings/project-settings.component.ts

```diff
@@ -180,15 +185,32 @@ export class ProjectSettingsComponent {
     ...updateWorkflowGroupsMutation(),
     onSuccess: () => {
       this.queryClient.invalidateQueries({ queryKey: getGroupsWithWorkflowsQueryKey({ path: { repositoryId: this.repositoryId() } }) });
+      this.messageService.add({ severity: 'success', summary: 'Success', detail: 'Workflow groups updated successfully' });
+    },
+    onError: (error: Error) => {
+      this.messageService.add({ severity: 'error', summary: 'Error', detail: String(error) });
     },
   }));
 
-  getWorkflowLabelOptions() {
-    return Object.values(WorkflowDtoSchema.properties.label.enum);
+  getWorkflowLabelOptions(currentLabel: string) {
+    const assignedLabels = this.workflows().map(wf => wf.label);
+    return Object.values(WorkflowDtoSchema.properties.label.enum).filter(label => {
+      const isDeployment = label === 'DEPLOYMENT' && assignedLabels.includes('DEPLOYMENT');
+      const isBuild = label === 'BUILD' && assignedLabels.includes('BUILD');
+      return label === currentLabel || (!isDeployment && !isBuild);
+    });
   }
 
   onChangeLabel(workflow: WorkflowDto) {
-    this.workflowLabelMutation.mutate({ path: { workflowId: workflow.id }, body: workflow.label });
+    const label = workflow.label;
+    if (label === 'DEPLOYMENT' || label === 'BUILD') {
+      const existingLabel = this.workflows().find(wf => wf.label === label && wf.id !== workflow.id);
+      if (existingLabel) {
+        console.warn(`Only one workflow can be labeled as ${label}.`);
```

Comment:
question(ux): maybe we should add a confirmation modal here to signal to the user, that the existing label is changed. There should be a component from primeng